File: apps/functions/datagate/src/main/scala/pknguyen/dategate/data_profiling.scala
package pknguyen.dategate

import org.apache.spark.sql.SparkSession

object data_profiling {
    def main(args: Array[String]): Unit = {
        println("[DataGate] Data Profiling...")
        
        val spark = SparkSession.builder()
            .appName("[DataGate] Data Profiling")
            .config()
            .getOrCreate()
        
        spark.stop()
    }
}

File: apps/functions/datagate/src/main/java/pknguyen/dategate/data_profiling.java
package pknguyen.dategate;

import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;

public class data_profiling {
    public static void main (String[] args) throws Exception {
        SparkSession spark = SparkSession.builder()
                .appName("[DataGate] Data profiling")
                .getOrCreate();
        try {
            System.out.println("[INFO] Start data profiling");
            String table = "rest.bronze.yellow_tripdate";
            Dataset<Row> df = spark.read().format("iceberg").load(table);
            System.out.println(df.count());
        } finally  {
            spark.stop();
        }
    }

}
